sklearn/svm/src/libsvm/libsvm_sparse_helper.c

```diff
@@ -1,5 +1,6 @@
 #include <stdlib.h>
-#include <numpy/arrayobject.h>
+#define PY_SSIZE_T_CLEAN
```

Comment:
Can you document why this is needed?

This comment also applies to other files.

Reply:
https://docs.python.org/3/c-api/intro.html#include-files
Do you want that included here as comment? It is THE PYTHON DOCUMENTATION!

Reply:
You are right. Ignore my comment.

Reply:
Actually, would would we use `Py_ssize_t` instead of our own `intp_t`?

Reply:
Or we could even use `intptr_t` from the `C99` / `C11` standard:

- https://www.man7.org/linux/man-pages/man3/uintptr_t.3type.html

The dev version of the numpy doc says that `intptr_t` is the C type equivalent of `np.intp`: https://numpy.org/devdocs/reference/arrays.scalars.html#numpy.intp

Reply:
Actually: https://peps.python.org/pep-0353/#why-not-py-intptr-t but I don't understand the meaning of this paragraph :)

Reply:
I remember that we had this discussion somewhere. In typedefs.pxd, we can now easily change that.
As the SVM/Liblinear code is C, I can't use our Cython tyedefs. I can use whatever makes you more happy: `Py_ssize`, `size_t`, `intpr_t`, see https://github.com/numpy/numpy/issues/24855.

Reply:
See more discussion here:

https://github.com/numpy/numpy/issues/1654#issuecomment-71082792

`Py_ssize_t` might be more correct but numpy always indexes with `numpy.intp` which is aligned with `intptr_t` instead. There should not be any difference on any supported architecture though.

I am fine with using `intp_t` in Cython files and `Python.h` with `Py_ssize_t` in C/C++ source files then.

We should probably document this in our `_typedefs.pxd` file.